google/cloud/aiplatform/metadata/schema/google/artifact_schema.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Dict
+
+from google.cloud.aiplatform.compat.types import artifact as gca_artifact
+from google.cloud.aiplatform.metadata.schema import base_artifact
+from google.cloud.aiplatform.metadata.schema import utils
+
+
+class VertexDataset(base_artifact.BaseArtifactSchema):
+    """An artifact representing a Vertex Dataset."""
+
+    SCHEMA_TITLE = "google.VertexDataset"
+
+    def __init__(
+        self,
+        dataset_id: Optional[str] = None,
```

Comment:
I'm confused by this argument. If this is provided shouldn't it populate the uri and resourceName automatically?

Also, the subclass says this is the artifact id for the Artifact not the resource id of the vertex resource. Perhaps we need to expose both.

Reply:
You are correct, I mixed up the artifact_id and resouce_id, I have updated the code to accept resouce_name for Vertex  artifacts such as VertexModel and VertexDataset. With that I'm populating the resouce_name accordingly. 

Also I'm renaming the resouce_id to artifact_id and execution_id for artifact and executions respectively.